app/hotels/src/navigation/singleHotel/SingleHotelMapNavigationScreen.js

```diff
@@ -6,28 +6,45 @@ import { withMappedNavigationAndConfigProps } from 'react-navigation-props-mappe
 
 import type { AvailableHotelSearchInput } from '../../singleHotel/AvailableHotelSearchInput';
 import SingleHotelMap from '../../map/singleHotel/SingleHotelMapScreen';
+import SingleHotelContext from './SingleHotelContext';
+import Stay22SingleHotelMapScreen from '../../map/singleHotel/Stay22SingleHotelMapScreen';
 
 type Props = {
   ...AvailableHotelSearchInput,
   currency: string,
 };
 
 class SingleHotelMapNavigationScreen extends React.Component<Props> {
-  render() {
+  renderInner = ({ apiProvider }) => {
+    const child =
+      apiProvider === 'booking' ? (
+        <React.Fragment>
+          <SingleHotelMap
+            search={{
+              hotelId: this.props.hotelId,
+              checkin: this.props.checkin,
+              checkout: this.props.checkout,
+              roomsConfiguration: this.props.roomsConfiguration,
+            }}
+            currency={this.props.currency}
+          />
+        </React.Fragment>
+      ) : (
+        <Stay22SingleHotelMapScreen />
```

Comment:
So booking.com hotel screen gets its props via navigation, but Stay22 does not need to pass down these props?

Reply:
If I would create the booking.com query today, I would probably just connect to the context as well. I could refactor it, so they both fetch from the context.

Reply:
Refactored so none of them takes props.